### PR TITLE
Implemented Redux in React Components

### PR DESCRIPTION
**Pull request by AsimKhan2019**
Implemented Redux in React Components for Books and Categories.

### CODE REVIEW

File: src/components/AddBook.js [hunk 1]
```diff
@@ -1,12 +1,27 @@
...
-    <input type="text" placeholder="Enter author name" id="author" />
-    <button type="button" text="Add" value="Add" label="Add">Add Book</button>
-  </div>
-);
+const AddBook = () => {
+  const dispatch = useDispatch();
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    console.log('Form submitted');
```
Comment (omarsalem7) [inline src/components/AddBook.js]: - [x] Kindly remove console.log from your code to make it clean.

**Review by omarsalem7 — CHANGES_REQUESTED** [review]
Hi @AsimKhan2019 😊, 

Good job so far! 👏
You have done well reaching this point of the milestone, but there are some issues that you still need to work on to go to the next project but you are almost there! 

you-are-near

## Highlights
- Add and remove book done with dispatch action correctly ✔️

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/AddBook.js (hunk 1 above)
Comment (AsimKhan2019) [reply src/components/AddBook.js]: @omarsalem7 Thank you for the feedback. I will remove this statement now.

**Review by t-yanick — APPROVED** [review]
Hi @AsimKhan2019, 

### Great job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Descriptive Readme file

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

### N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by AsimKhan2019** [comment]
@t-yanick Thank you for the approval

**Pull request merged**